packages/core/src/browser/shell/description-widget.ts

```diff
@@ -0,0 +1,29 @@
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+ import { Emitter } from '../../common/event';
+
+export interface DescriptionWidget {
+    description: string;
+    onDescriptionChange: Emitter<void>;
+}
+
+export namespace DescriptionWidget {
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    export function is(arg: any): arg is DescriptionWidget {
+        return arg !== undefined && !!arg['onDescriptionChange'];
+    }
```

Comment:
I wasn't so sure the file was necessary, and instead it could be directly added to the `tree-view`.
Adding it under `shell/` is also a bit awkward.

Reply:
I'm now moved this file into `view-container.ts` as it only really applies to the container widgets